backend/models.py

```
@@ -272,10 +279,20 @@ def __init__(self, text, order, module_id):
 
     def validate_module_type(self):
         module = Module.query.get(self.module_id)
-        if module and module.module_type != "challenge":
+        if module and module.module_type not in [ModuleType.CHALLENGE, ModuleType.BONUS_CHALLENGE]:
             raise ValueError("Associated module must be of type 'challenge'")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)